paperqa/contrib/zotero.py

```diff
@@ -184,7 +175,14 @@ def iterate(
         while num_remaining > 0:
             cur_limit = min(max_limit, num_remaining)
             self.logger.info(f"Downloading new batch of up to {cur_limit} papers.")
-            _items = self.top(**query_kwargs, limit=cur_limit, start=i)
+
+            if collection_name: 
+                collection_id = self._get_collection_id(collection_name)
+                if not collection_id: continue
```

Comment:
What does this part do?

Reply:
```suggestion
        collection_id = None
        if collection_name: 
            collection_id = self._get_collection_id(collection_name)

        while num_remaining > 0:
          ....
          ....
            if collection_id: 
                _items = self.collection_items(collection_id, limit=cur_limit, start=i)
            else:
                _items = self.top(**query_kwargs, limit=cur_limit, start=i)    
```

Reply:
That looks like it could work. But I think something more reasonable is to just raise an error if the collection can't be found. It's less obvious what is going on if it just starts returning papers from the entire library when I type the collection name incorrectly.

Reply:
Thanks for your suggestions!!  The new commit: [8dc4b50](https://github.com/whitead/paper-qa/pull/48/commits/8dc4b502cb1557f87c3254dc3ebec703892f5036)

Reply:
Thanks, looks good. I have some more comments below